### PR TITLE
Resolve trigger dagrun operator deprecations in tests

### PR DESCRIPTION
**Pull request by dirrao**
Resolve trigger dagrun operator deprecations in tests
related: https://github.com/apache/airflow/issues/38642

### CODE REVIEW

**Comment by potiuk** [comment]
conflicts :(

**Comment by dirrao** [comment]
> conflicts :(

Done

**Review by jscheffl — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]

**Review by Lee-W — APPROVED** [review]
One minor CI failure, but everything else looks good

File: tests/operators/test_trigger_dagrun.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 import pendulum
 import pytest
 
-from airflow.exceptions import AirflowException, DagRunAlreadyExists, TaskDeferred
+from airflow.exceptions import AirflowException, DagRunAlreadyExists, TaskDeferred, RemovedInAirflow3Warning
```
Comment (uranusjr) [inline tests/operators/test_trigger_dagrun.py]:
```suggestion
from airflow.exceptions import AirflowException, DagRunAlreadyExists, RemovedInAirflow3Warning, TaskDeferred
```

**Review by uranusjr — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Pull request merged**